compiler/rustc_trait_selection/src/autoderef.rs

```diff
@@ -222,19 +222,10 @@ pub fn report_autoderef_recursion_limit_error<'tcx>(tcx: TyCtxt<'tcx>, span: Spa
         Limit(0) => Limit(2),
         limit => limit * 2,
     };
-    struct_span_err!(
-        tcx.sess,
+    tcx.sess.emit_err(AutoDerefReachedRecursionLimit {
         span,
-        E0055,
-        "reached the recursion limit while auto-dereferencing `{:?}`",
-        ty
-    )
-    .span_label(span, "deref recursion limit reached")
-    .help(&format!(
-        "consider increasing the recursion limit by adding a \
-             `#![recursion_limit = \"{}\"]` attribute to your crate (`{}`)",
+        ty: format!("{:?}", ty),
```

Comment:
How does the error output change if we don't use the `debug` output for `ty` here?

Reply:
This wasn't needed, someone implemented `IntoDiagnosticArg` for `Ty` already :)

I changed the commit to implement `IntoDiagnosticArg` for `Unevaluated` too.